### PR TITLE
Docs: inline code for lint function

### LINKED ISSUE #712: Indicate that inline-data can be used in lintr::lint(filename = "...")
As noted in https://github.com/jimhester/lintr/issues/474 there is no indication in the docs for `lint` that inline-data can be used as input to the function (or how the inline data should be formatted).

### PR DESCRIPTION
**Pull request by russHyde**
Fixes #712 and #474

Add description of use of inline R code in the filename argument of `lint()`

### CODE REVIEW

File: R/lint.R [hunk 1]
```diff
@@ -12,15 +12,27 @@ NULL
...
 #'
 #' Apply one or more linters to a file and return the lints found.
+#'
 #' @name lint_file
-#' @param filename the given filename to lint.
-#' @param linters a named list of linter functions to apply see \code{\link{linters}}
-#' for a full list of default and available linters.
+#'
+#' @param filename either the filename for a file to lint, or a character
+#' string of inline R code for linting. If a newline character is detected,
```
Comment (MichaelChirico) [inline R/lint.R:19]: nit: "if" should be "if and only if". I might reword it to something like

> The latter [inline data] applies whenever `filename` has a newline character (`\n`).

Comment (russHyde) [reply R/lint.R]: Fixed

**Review by MichaelChirico — APPROVED** [review]
LGTM!
